Added image upload feature

HetaviM29 opened the pull request:
Added image upload , download , and post a review feature.

HetaviM29 committed:
modified:   app.js
	modified:   models/imagemodel.js
	modified:   views/index.ejs

HetaviM29 committed:
modified:   app.js

HetaviM29 committed:
modified:   app.js

HetaviM29 committed:
modified:   app.js

HetaviM29 committed:
modified:   .gitignore
	modified:   app.js
	new file:   controllers/imageController.js
	modified:   package-lock.json
	modified:   package.json
	new file:   routes/imageRoute.js

cpandey05 commented on controllers/imageController.js:
```diff
@@ -0,0 +1,109 @@
+
+const uploadImage = async (req, res) => {
+    try {
+        if (!req.file) {
```
Do you observe a lot of repetition in code?

cpandey05 commented on controllers/imageController.js:
```diff
@@ -0,0 +1,109 @@
+const downloadImage = async (req, res) => {
+    try {
+        const file = path.join(__dirname, '../public/images', req.params.filename);
+        res.download(file);
```
why controller knows about image path

HetaviM29 replied:
Fixed the repetition of the code . 
Also removed the image path from the controller. Modified the code by added service layer.

HetaviM29 committed:
modified:   controllers/imageController.js
	new file:   dtos/imageDTO.js
	new file:   services/imageService.js

HetaviM29 committed:
modified:   app.js
	modified:   controllers/imageController.js

HetaviM29 committed:
modified:   controllers/imageController.js
	modified:   services/imageService.js

HetaviM29 committed:
new file:   Real Review.postman_collection.json

HetaviM29 commented:
1.  Web app deployed on EC2: http://16.16.167.244:3000/
2. Postman collection added at postman/RealReview.postman_collection.json
3. Note: The POST /images (file upload) route must be tested manually — Postman Runner does not support file uploads.

HetaviM29 committed:
modified:   dtos/imageDTO.js
	modified:   models/imagemodel.js
	modified:   services/imageService.js
	modified:   views/index.ejs

cpandey05 commented:
Do not see any S3 code there

HetaviM29 commented:
Sir still working on it as there are few errors which are still not getting resolved will upload by today itself . 
Also sir can you please tell apart from S3 with lifecycle rules is there anything else that we need to do?

HetaviM29 committed:
modified:   controllers/imageController.js
	modified:   dtos/imageDTO.js
	modified:   models/imagemodel.js
	modified:   package-lock.json
	modified:   package.json
	modified:   routes/imageRoute.js
	modified:   services/imageService.js
	modified:   views/index.ejs

HetaviM29 committed:
modified:   routes/imageRoute.js

HetaviM29 committed:
modified:   controllers/imageController.js
	modified:   models/imagemodel.js
	modified:   services/imageService.js
	modified:   views/index.ejs

HetaviM29 committed:
modified:   controllers/imageController.js
	modified:   routes/imageRoute.js
	modified:   views/index.ejs

HetaviM29 commented:
Updated ip to run on port 80 http://13.51.159.68/
Also integrated S3 with Iam role , images are getting uploaded on S3 but due to some issue they are not getting fetched in the frontend tried all the possible ways but nothing seems to work , also if I run code on local host then everything works perfectly images get uploaded on S3  ,  reviews in DB , frontend is also working on local host .

cpandey05 commented on services/imageService.js:
```diff
@@ -0,0 +1,36 @@
+
+const getAllS3Images = async () => {
+    const command = new ListObjectsV2Command({
+        Bucket: process.env.AWS_S3_BUCKET,
```
What approach are you relying on for discovering creds

HetaviM29 replied:
currently i am using dotenv file to store credentials

HetaviM29 replied:
Fixed the image fetching issue : http://13.51.159.68/ 
So now all the images are getting uploaded to S3 reviews are getting uploaded in MongoDB as of now .

HetaviM29 committed:
modified:   controllers/imageController.js

HetaviM29 committed:
modified:   controllers/imageController.js

HetaviM29 committed:
modified:   controllers/imageController.js

HetaviM29 committed:
new file:   config/dynamodb.js
	modified:   controllers/imageController.js
	modified:   package-lock.json
	modified:   package.json
	modified:   services/imageService.js
	modified:   views/index.ejs

HetaviM29 committed:
Create README.md

HetaviM29 committed:
modified:   app.js
	modified:   controllers/imageController.js
	deleted:    models/imagemodel.js
	modified:   views/index.ejs

HetaviM29 committed:
Merge branch 'feature/image-upload' of https://github.com/HetaviM29/Real-review2 into feature/image-upload

HetaviM29 commented:
Moved Database from mongodb to dynamodb .
Also S3 has lifecycle rule updated for deletion of images after 5 days using lambda function and synced it with dynamodb so after 5 days meta data associated with image also gets deleted. 
http://13.51.159.68/

HetaviM29 committed:
modified:   services/imageService.js

HetaviM29 committed:
modified:   controllers/imageController.js

cpandey05 commented on package.json:
```diff
@@ -10,7 +10,10 @@
+    "@aws-sdk/client-dynamodb": "^3.823.0",
     "@aws-sdk/client-s3": "^3.817.0",
+    "@aws-sdk/lib-dynamodb": "^3.823.0",
+    "@aws-sdk/s3-request-presigner": "^3.823.0",
```
Do we need two Dynamo DB dependencies?

cpandey05 commented on services/imageService.js:
```diff
@@ -1,36 +1,61 @@
-    // Returns { filename, reviews } for all reviewed images
-    return await Review.find({}, { filename: 1, reviews: 1, _id: 0 });
+    const data = await docClient.send(new ScanCommand({ TableName: "Reviews" }));
+    return data.Items || [];
```
Why Scan command?

HetaviM29 replied:
It is used to fetch all the reviews from the table
